data-entry/dataEntry.app.js

```diff
@@ -78,12 +78,62 @@
                         if (key.simple) {
                             (function(key) {
                                 key.getDomainValues().then(function success(values) {
-                                    dataEntryModel.domainValues[key.colset.columns[0].name] = [];
-                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name];
-
-                                    angular.forEach(values.data, function(value) {
-                                        var field = Object.keys(value)[0];
-                                        domainValues.push(value[field]);
+                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name] = [];
+                                    var table = key.key._table;
+                                    var annotations = table.annotations.all();
+                                    var termColumnName = null;
+
+
+                                    // TODO need to figure out how to handle if mutiple annotations are present
+                                    angular.forEach(annotations, function(annotation) {
+                                        // there should only be one term annotation
+                                        if (annotation.content.term) {
+                                            termColumnName = annotation.content.term;
+                                        }
+                                    });
+                                    // if a term annotation is defined
+                                    if (termColumnName) {
+                                        // do nothing and break out of "if"
+                                    }
+                                    // use the term column instead
+                                    else if(table.columns._columns.Term) {
```

Comment:
Use square bracket lookups when looking up based on user data. This is a little gray area here because technically annotations are also defined by us. But let's treat them like user data. So do not use "dot" (`.`) lookups.